tests/test-class-wpseo-sitemaps.php

```diff
@@ -122,4 +156,172 @@ public function test_main_sitemap() {
 			'Served from transient cache',
 		) );
 	}
+
+	/**
+	 * Exclude user from sitemaps by excluding the entire role
+	 */
+	public function test_author_exclusion_from_sitemap_by_role() {
+		$user = $this->get_user();
+
+		// Filter out all administrators.
+		add_filter( 'pre_option_wpseo_xml', array( $this, 'filter_exclude_author_by_administrator_role' ) );
+
+		$result = self::$class_instance->user_sitemap_remove_excluded_authors( array( $user ) );
+
+		// User should be removed.
+		$this->assertEquals( $result, array() );
+	}
+
+	/**
+	 * Test if a user is excluded from sitemaps when disabled on profile
+	 */
+	public function test_author_exclusion_from_sitemap_by_preference() {
+		$user = $this->get_user();
+
+		// Enable author sitemaps.
+		add_filter( 'pre_option_wpseo_xml', array( $this, 'filter_enable_author_sitemaps' ) );
+
+		// Make sure the user has posts.
+		add_filter( 'get_usernumposts', array( $this, 'filter_user_has_posts' ) );
+
+		// Add filter to exclude the user.
+		add_filter( 'get_the_author_wpseo_excludeauthorsitemap',
+			array( $this, 'filter_user_meta_exclude_author_from_sitemap' ) );
+
+		$result = self::$class_instance->user_sitemap_remove_excluded_authors( array( $user ) );
+
+		// User should be removed
+		$this->assertEquals( $result, array() );
+	}
+
+	/**
+	 * Test if a user is excluded from the sitemap when there are no posts
+	 */
+	public function test_author_exclusion_from_sitemap_by_zero_posts() {
+		$user = $this->get_user();
+
+		// Don't allow no posts.
+		add_filter( 'pre_option_wpseo_xml', array( $this, 'filter_exclude_author_by_no_posts' ) );
+
+		// Make the user have -no- posts.
+		add_filter( 'get_usernumposts', array( $this, 'filter_user_has_no_posts' ) );
+
+		$result = self::$class_instance->user_sitemap_remove_excluded_authors( array( $user ) );
+
+		// User should be removed.
+		$this->assertEquals( $result, array() );
+	}
+
+	/**
+	 * Test if a user is excluded from the sitemap when there are no posts
```

Comment:
The docs here are exactly the same for the above test, yet there are differences in implementation, so I'm assuming you're testing something slightly different here. Could you please elaborate in the function comment?